packages/@webex/plugin-meetings/src/recording-controller/index.ts

```diff
@@ -0,0 +1,214 @@
+import PermissionError from '../common/errors/permission';
+import {CONTROLS, HTTP_VERBS} from '../constants';
+import LocusInfo from '../locus-info';
+import MeetingRequest from '../meeting/request';
+import MeetingUtil from '../meeting/util';
+import RecordingAction from './enums';
+import Util from './util';
+import LoggerProxy from '../common/logs/logger-proxy';
+
+/**
+ * @description Recording manages the recording functionality of the meeting object, there should only be one instantation of recording per meeting
+ * @export
+ * @private
+ * @class Recording
+ */
+export default class RecordingController {
+  /**
+   * @instance
+   * @type {MeetingRequest}
+   * @private
+   * @memberof RecordingController
+   */
+  private request: MeetingRequest;
+  /**
+   * @instance
+   * @type {MeetingRequest}
+   * @private
+   * @memberof RecordingController
+   */
+  private info: LocusInfo;
+
+  /**
+   * @instance
+   * @type {string}
+   * @private
+   * @memberof RecordingInfo
+   */
+  private serviceUrl: string;
+
+  /**
+   * @instance
+   * @type {string}
+   * @private
+   * @memberof RecordingInfo
+   */
+  private sessionId: string;
+
+  /**
+   * @instance
+   * @type {string}
+   * @private
+   * @memberof RecordingInfo
+   */
+  private locusUrl: string;
+
+  /**
+   * @instance
+   * @type {string}
+   * @private
+   * @memberof RecordingInfo
+   */
+  private locusId: string;
+
+  /**
+   * @param {MeetingRequest} request
+   * @param {LocusInfo} info
+   * @constructor
+   * @memberof RecordingController
+   */
+  constructor(request: MeetingRequest, info: LocusInfo) {
+    this.initialize(request, info);
+  }
+
+  /**
+   * @param {MeetingRequest} request
+   * @param {LocusInfo} info
+   * @returns {void}
+   * @private
+   * @memberof RecordingController
+   */
+  private initialize(request: MeetingRequest, info: LocusInfo) {
+    this.request = request;
+    this.info = info;
```

Comment:
info is being save here, but not in the set() method. We should either save it in both places or not save it at all. I prefer to not save it at all.

Reply:
yeah (as above) we won't use locusInfo anylonger.